migration/migrate.go

```diff
@@ -0,0 +1,94 @@
+package migration
+
+import (
+	"encoding/json"
+	liberr "github.com/konveyor/controller/pkg/error"
+	"github.com/konveyor/tackle2-hub/database"
+	"github.com/konveyor/tackle2-hub/model"
+	"gorm.io/gorm"
+)
+
+//
+// Migrate the hub by applying all necessary Migrations.
+func Migrate(migrations []Migration) (err error) {
+	var db *gorm.DB
+	db, err = database.Open(false)
+	if err != nil {
+		return
+	}
+
```

Comment:
I wonder if we don't want something like:
```
defer func() {
    if err != nil {
        _ = database.Close(db)
    }
}
```
here to prevent leaking db connections on error?

The `var db *gorm.DB` on line 47 would need to be removed so the loop is just re-assigning the outer `db`.